homeassistant/components/zone/__init__.py

```diff
@@ -106,6 +110,29 @@ def async_active_zone(
     min_dist = None
     closest = None
 
+    # Update state of current_zone
+    if current_zone:
+        current_zone = hass.states.get(current_zone.entity_id)
+
+    if (
+        current_zone
+        and current_zone.state != STATE_UNAVAILABLE
+        and not current_zone.attributes.get(ATTR_PASSIVE)
+    ):
+        zone_dist = distance(
+            latitude,
+            longitude,
+            current_zone.attributes[ATTR_LATITUDE],
+            current_zone.attributes[ATTR_LONGITUDE],
+        )
+        if zone_dist is not None:
+            # Still within zone if zone and location circles intersect
+            within_zone = zone_dist < current_zone.attributes[ATTR_RADIUS] + radius
```

Comment:
Do we want to test for `<` or `<=` here?
Since it's floating point maths it's essentially the same for all practical purposes, but maybe one or the other is easier to reason about?

Reply:
If only looking on this case I'd use `<=` but if there are other cases that depend on the same measurements we should consider them too so all cases are compatible and make sense.